web/packages/teleterm/src/services/tshd/createClient.ts

```diff
@@ -648,6 +651,24 @@ export default function createClient(
         });
       });
     },
+
+    updateHeadlessAuthenticationState(
+      params: UpdateHeadlessAuthenticationStateParams
+    ) {
+      const req = new api.UpdateHeadlessAuthenticationStateRequest()
```

Comment:
It should be wrapped in `withAbort` (like [here](https://github.com/gravitational/teleport/blob/d58f364bf43dbc2243caccdc38b275b38ad658bd/web/packages/teleterm/src/services/tshd/createClient.ts#L382)) and you should pass `abortSignal` to the function, otherwise it won't work.